server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -394,24 +397,25 @@ public void onResponse(RolloverResponse result) {
 
                 @Override
                 public void onFailure(Exception e) {
-                    failRequestsWhenPrerequisiteActionFailed(dataStream, bulkRequest, responses, e);
```

Comment:
Hi @nielsbauman,

Thank you for pointing that out. In your suggested solution, we set exceptions in the BulkRequest for indices. However, since we have separate sets for dataStreamsToBeRolledOver and failureStoresToBeRolledOver, I believe the issue you mentioned may not be present in the current code.

That said, I could be mistaken, and if so, I'd appreciate your guidance on the changes I need to make.